pkg/webhook/preflight/nutanix/imagekubernetesversioncheck.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2025 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nutanix
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	vmmv4 "github.com/nutanix/ntnx-api-golang-clients/vmm-go-client/v4/models/vmm/v4/content"
+
+	carenv1 "github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/api/v1alpha1"
+	"github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pkg/webhook/preflight"
+)
+
+// The regex captures the Kubernetes version in the format of 1.x.y, where x and y are digits.
+// Example: kubedistro-rocky-9.5-vgpu-1.32.3-20250604180644 -> 1.32.3.
+var kubernetesVersionRegex = regexp.MustCompile(`(?i)\b[vV]?(1\.\d+(?:\.\d+)?)\b`)
+
+type imageKubernetesVersionCheck struct {
+	machineDetails    *carenv1.NutanixMachineDetails
+	field             string
+	nclient           client
+	clusterK8sVersion string
+}
+
+func (c *imageKubernetesVersionCheck) Name() string {
+	return "NutanixVMImageKubernetesVersion"
+}
+
+func (c *imageKubernetesVersionCheck) Run(ctx context.Context) preflight.CheckResult {
+	if c.machineDetails.ImageLookup != nil {
+		return preflight.CheckResult{
+			Allowed:  true,
+			Warnings: []string{fmt.Sprintf("%s uses imageLookup, which is not yet supported by checks", c.field)},
+		}
+	}
+
+	if c.machineDetails.Image != nil {
+		images, err := getVMImages(c.nclient, c.machineDetails.Image)
+		if err != nil {
+			return preflight.CheckResult{
+				Allowed: false,
+				Error:   true,
+				Causes: []preflight.Cause{
+					{
+						Message: fmt.Sprintf("failed to get VM Image: %s", err),
+						Field:   c.field,
+					},
+				},
+			}
+		}
+
+		if len(images) == 0 {
+			return preflight.CheckResult{
+				Allowed:  true,
+				Warnings: []string{"expected to find 1 VM Image, found none"},
+			}
+		}
+
+		if err := c.checkKubernetesVersion(&images[0]); err != nil {
+			return preflight.CheckResult{
+				Allowed: false,
+				Error:   true,
+				Causes: []preflight.Cause{
+					{
+						Message: err.Error(),
```

Comment:
Setting `Error` to true means there was an internal error.
 
Having read through `checkKubernetesVersion`, I see nothing that should be presented as an internal error.